status-im/status-react
# fix: no routes are found in case of valid amount in fiat is entered

fixes #18561 
fixes #19999

### Summary

This PR fixes a bug that makes no routes being fetched when the users switch the input mode to display values in fiat and a valid amount in fiat is entered. Also fixed some minor but noticeable bugs that appeared after fixing the main bug:

- Visual overflow of the suffix text of the input when swapping mode to fiat and the number is too big. Needed to refactor a bit the views and styles of Token Input component.
- Don't allow the user to use more decimals than the token supports.
- Fiat character symbol was not showing in the conversion.
- Fiat character symbol was not used when showing the max amount.
- Many trailing zeroes issues when leading with large numbers.
- Routes being fetched when swapping to fiat, and vise versa

#### Demo

https://github.com/status-im/status-mobile/assets/18485527/c67ae017-b134-4813-b716-99ba9167ef8f

#### Platforms

- Android
- iOS

#### Areas that maybe impacted

##### Functional

- wallet / transactions

### Steps to test

- Open Status
- Login
- Go to wallet
- Select an account
- Tap on Send button
- Select an asset where the user has funds on
- Tap on switch button on the input to display values in fiat
- Enter a valid amount in fiat
- Wait for routes to be loaded
- Verify correct behavior of input and routes loading

status: ready
